### ISSUE TITLE
cmake 3.17.x can't build darktable

### ISSUE THREAD

**Issue by st-binder** [darktable-org/darktable#4672]
darktable version 3.0.1
cmake 3.17.1
platform: amd, archlinux 64bit

it isn't possible to build darktable, cmake stops with error message. downgrading cmake to 3.16.5 solves the problem, but this isn't a solution.

CMAKE console output:

```
CMake Error at data/kernels/CMakeLists.txt:34 (foreach):
  Unknown argument:
    /share/local/software/asp/darktable/src/darktable-3.0.1/data/kernels/atrous.cl
CMake Error at data/kernels/CMakeLists.txt:35 (testcompile_opencl_kernel):
  testcompile_opencl_kernel Macro invoked with incorrect arguments for macro
  named: testcompile_opencl_kernel
CMake Error at data/kernels/CMakeLists.txt:36 (endforeach):
  endforeach An ENDFOREACH command was found outside of a proper FOREACH
  ENDFOREACH structure.  Or its arguments did not match the opening FOREACH
  command.

```

**Comment by codingdave** [comment]
I cannot confirm your building issues with CMake 3.17.0 on manjaro.

$ lsb_release -a
LSB Version:	n/a
Distributor ID:	ManjaroLinux
Description:	Manjaro Linux
Release:	19.0.2
Codename:	Kyria
Kernel: Linux 5.5.13-1-MANJARO
(Architecture: x86-64)
$ cmake -version
cmake version 3.17.0

**Comment by st-binder** [comment]
so i think it's related to archlinux only.

**Comment by coffee-lord** [comment]
@st-binder The issue you're describing isn't distro-specific. It was fixed in 767d48e0f60e7f858e8b31a88dd8cc1258e7ee9e but the fix didn't make it into 3.0.1 AFAIK.
